pkg/endpoint/events.go

```
@@ -300,10 +299,12 @@ func (ev *EndpointPolicyBandwidthEvent) Handle(res chan interface{}) {
 	if bandwidthEgress != "" {
 		bps, err = bandwidth.GetBytesPerSec(bandwidthEgress)
 		if err == nil {
-			err = bwmap.Update(e.ID, bps)
+			ev.bwm.UpdateBandwidthLimit(e.ID, bps)
+		} else {
+			e.getLogger().WithError(err).Debugf("failed to parse bandwidth limit %q", bandwidthEgress)
 		}
 	} else {
-		err = bwmap.SilentDelete(e.ID)
```

This line made me think: with the new reconciliation mechanism, there is no way to specify ignoreMissing, and we always increase the error metric whenever statedb tries to remove a non-existent entry. While the latter makes sense to me, probably we'd also need to increase the error metric ourselves at DeleteBandwidthLimit if it were a regular Delete (not SilentDelete) here.

---

Ah yes! Excellent observation. I thought I had used SilentDelete in ops_linux.go, but now that I checked it I hadn't, so now it'll forever retry deletions if the entry isn't there, which obviously is wrong. I'll fix this!

I think in the general case we're probably fine with ignoring the error from a delete of a non-existing entry. Though of course we could make it configurable, but not quite sure when we'd want to retry failed deletes (though we can log a warning perhaps?).